website/source/docs/concepts/tokens.html.md

```diff
@@ -42,14 +42,53 @@ Finally, the `-policy` flag can be used to set the policies associated
 with the token. Learn more about policies on the
 [policies concepts](/docs/concepts/policies.html) page.
 
+<a name="ttls-and-leases" />
 ## Token Time-To-Live and Leases
 
-Every non-root token has a time-to-live (TTL) associated with it.  After the
-TTL is up, the token will no longer function. In addition to no longer
-functioning, Vault will revoke it.
+Every non-root token has a time-to-live (TTL) associated with it, which is
+the number of seconds the token is valid for, _measured from the moment it is
+created_. After the TTL is up, the token will no longer function, and
```

Comment:
> Under the hood TTL represents "seconds from issue", and only changes when you renew the token, but users see it as "seconds remaining", and it's constantly decreasing.

This isn't _quite_ right. Under the hood there isn't any real TTL being stored; it's represented by a timer that on expiration revokes the token/lease. Seconds from issue is really only used to calculate the total duration that has passed since issuance and compare that to the max TTL. 

So, here's the current state: a token has:

* creation time: for info and also for comparing against the max TTL
* creation ttl: in the code (the `TokenEntry` struct) this is the TTL field (for backwards compatibility) but in the token lookup it's the `creation_ttl` because it is never changed outside of initial token creation. This is really only exposed in a lookup call for informational purposes; in theory it can be used to detect whether a renew ever took place, but because the creation ttl is often the same as the max TTL a renew could happen with a shorter increment that decreases the current lifetime of the token, without affecting the value exposed here. This is why there's also a last renewal time exposed at lookup time.

The way that these are exposed in a lookup call is that the creation timestamp is a straight map, the TTL from the TokenEntry turns into the creation TTL, and the TTL field that comes back from a lookup call is actually sourced from the expiration manager keeping track of the lifetime. The point here is that the TTL you see in a lookup call that's always decreasing is actually inferred; it's not actually a property of the token itself. In the same way, the max TTL is not a property of the token; it's inferred from the current max TTL set on the mount or in the system and from the creation time.